pkg/kmm/preflightvalidationocp_test.go

```diff
@@ -0,0 +1,329 @@
+		return builder, errors.New("error")
+	})
+
+	assert.Equal(t, "error", testBuilder.errorMsg)
+}
+
+func TestPullPreflightValidationOCP(t *testing.T) {
+	testCases := []struct {
+		name                string
+		namespace           string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
+	}{
+		{
+			name:                "test",
+			namespace:           "testns",
+			expectedError:       false,
+			addToRuntimeObjects: true,
+			expectedErrorText:   "",
+		},
+		{
+			name:                "",
+			namespace:           "testns",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "preflightvalidationocp 'name' cannot be empty",
+		},
+		{
+			name:                "test",
+			namespace:           "",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "preflightvalidationocp 'nsname' cannot be empty",
+		},
+		{
+			name:                "test",
+			namespace:           "testns",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "preflightvalidationocp object test doens't exist in namespace testns",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testPreflight := generatePreflight(testCase.name, testCase.namespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testPreflight)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		builderResult, err := PullPreflightValidationOCP(testSettings, testCase.name, testCase.namespace)
+
+		if testCase.expectedError {
+			assert.NotNil(t, err)
+
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, err.Error())
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.NotNil(t, builderResult)
+			assert.Equal(t, testCase.name, builderResult.Definition.Name)
+			assert.Equal(t, testCase.namespace, builderResult.Definition.Namespace)
+		}
+	}
+}
+
+func TestPreflightCreate(t *testing.T) {
+	testCases := []struct {
+		addToRuntimeObjects bool
+	}{
+		{
+			addToRuntimeObjects: true,
+		},
+		{
+			addToRuntimeObjects: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+		)
+
+		testPreflight := generatePreflight(defaultPreflightName, defaultPreflightNamespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testPreflight)
+		}
+
+		testBuilder := buildPreflightTestBuilderWithFakeObjects(runtimeObjects)
+
+		builderResult, err := testBuilder.Create()
+		assert.Nil(t, err)
+		assert.NotNil(t, builderResult)
+		assert.Equal(t, defaultPreflightName, builderResult.Definition.Name)
+		assert.Equal(t, defaultPreflightNamespace, builderResult.Definition.Namespace)
+	}
+}
+func TestPreflightUpdate(t *testing.T) {
```

Comment:
Please refer to:
https://github.com/openshift-kni/eco-goinfra/blob/c2a8833ad5034f4d11c21752928abe93e476350c/pkg/argocd/argocd_test.go#L244

Reply:
Done, Thanks